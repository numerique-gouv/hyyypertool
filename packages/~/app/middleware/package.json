{
  "name": "@~/app.middleware",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./session": {
      "default": "./src/session/index.ts"
    },
    "./vip_list.guard": {
      "default": "./src/vip_list.guard.tsx"
    },
    "./*": {
      "default": "./src/*.ts"
    }
  },
  "scripts": {
    "test": "bun test"
  },
  "dependencies": {
    "@~/app.core": "workspace:*",
    "@~/app.sentry": "workspace:*",
    "@~/app.ui": "workspace:*",
    "@~/app.urls": "workspace:*",
    "@~/moncomptepro.database": "workspace:*",
    "@~/moncomptepro.lib": "workspace:*",
    "hono": "4.6.17",
    "hono-sessions": "0.7.2",
    "openid-client": "5.7.0"
  },
  "devDependencies": {
    "@~/config.typescript": "workspace:*"
  }
}
