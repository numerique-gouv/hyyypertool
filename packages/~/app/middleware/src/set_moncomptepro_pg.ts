//

import type { MonComptePro_PgDatabase, Pool } from "@~/moncomptepro.database";
import type { Env, MiddlewareHandler } from "hono";

//

export function set_moncomptepro_pg(
  client: MonComptePro_PgDatabase,
): MiddlewareHandler<MonComptePro_Pg_Context> {
  return async function set_moncomptepro_pg_middleware({ set }, next) {
    set("moncomptepro_pg", client);
    await next();
  };
}
export function set_moncomptepro_pg_client(
  client: InstanceType<typeof Pool>,
): MiddlewareHandler<MonComptePro_Pg_Client_Context> {
  return async function set_moncomptepro_pg_middleware({ set }, next) {
    set("moncomptepro_pg_client", client);
    await next();
  };
}
//

export interface MonComptePro_Pg_Context extends Env {
  Variables: {
    moncomptepro_pg: MonComptePro_PgDatabase;
  };
}

export interface MonComptePro_Pg_Client_Context extends Env {
  Variables: {
    moncomptepro_pg_client: InstanceType<typeof Pool>;
    moncomptepro_pg: MonComptePro_PgDatabase;
  };
}
