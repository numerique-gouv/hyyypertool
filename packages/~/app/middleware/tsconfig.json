{
  "compilerOptions": {
    "jsxImportSource": "hono/jsx",
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/base/tsconfig.json",
  "references": [
    { "path": "../../infra/moncomptepro/database/tsconfig.json" },
    { "path": "../../infra/moncomptepro/lib/tsconfig.json" },
    { "path": "../core/tsconfig.json" },
    { "path": "../di/tsconfig.json" },
    { "path": "../sentry/tsconfig.json" },
    { "path": "../ui/tsconfig.json" },
    { "path": "../urls/tsconfig.json" }
  ]
}
