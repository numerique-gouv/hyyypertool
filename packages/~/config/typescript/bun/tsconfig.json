{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "extends": "../base/tsconfig.json",
  "compilerOptions": {
    "types": ["bun-types"]
  }
}
