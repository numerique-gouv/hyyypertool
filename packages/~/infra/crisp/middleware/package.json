{
  "name": "@~/crisp.middleware",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    ".": {
      "default": "./src/index.ts"
    }
  },
  "scripts": {
    "test": "bun test"
  },
  "dependencies": {
    "hono": "4.6.14"
  },
  "devDependencies": {
    "@~/config.typescript": "workspace:*",
    "@~/crisp.lib": "workspace:*",
    "bun-types": "1.1.38"
  }
}
