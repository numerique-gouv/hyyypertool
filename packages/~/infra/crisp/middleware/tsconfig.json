{
  "compilerOptions": {
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/bun/tsconfig.json",
  "references": [{ "path": "../lib/tsconfig.json" }]
}
