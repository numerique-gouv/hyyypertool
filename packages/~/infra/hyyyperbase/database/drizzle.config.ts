//

import type { Config } from "drizzle-kit";
import { env } from "node:process";

//

export default {
  dbCredentials: {
    url:
      env["DATABASE_URL"] ||
      "postgresql://postgres:postgres@localhost:6543/postgres?schema=public",
  },
  dialect: "postgresql",
  introspect: { casing: "preserve" },
  out: "src/drizzle",
  schema: "src/drizzle/schema.ts",
  strict: true,
  verbose: true,
} satisfies Config;
