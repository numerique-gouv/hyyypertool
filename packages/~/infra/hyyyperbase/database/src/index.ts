export { drizzle } from "drizzle-orm/node-postgres";
