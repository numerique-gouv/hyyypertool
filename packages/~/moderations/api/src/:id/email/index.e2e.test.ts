//

import { set_config } from "@~/app.middleware/set_config";
import { set_moncomptepro_pg } from "@~/app.middleware/set_moncomptepro_pg";
import { set_nonce } from "@~/app.middleware/set_nonce";
import { set_userinfo } from "@~/app.middleware/set_userinfo";
import {
  create_adora_pony_moderation,
  create_adora_pony_user,
  create_unicorn_organization,
} from "@~/moncomptepro.database/seed/unicorn";
import { empty_database, migrate, pg } from "@~/moncomptepro.database/testing";
import { beforeAll, beforeEach, expect, test } from "bun:test";
import { Hono } from "hono";
import app from "./index";

//

beforeAll(migrate);
beforeEach(empty_database);

//

test("GET /moderations/:id/email", async () => {
  await create_unicorn_organization(pg);
  await create_adora_pony_user(pg);

  const moderation_id = await create_adora_pony_moderation(pg, { type: "" });

  //

  const response = await new Hono()
    .use(set_config({ ALLOWED_USERS: "good@captain.yargs" }))
    .use(set_moncomptepro_pg(pg))
    .use(set_nonce("nonce"))
    .use(set_userinfo({ email: "good@captain.yargs" }))
    .route("/:id/email", app)
    .onError((error) => {
      throw error;
    })
    .request(`/${moderation_id}/email`);

  if (response.status >= 400) throw await response.text();

  expect(response.status).toBe(200);
  expect(await response.text()).toMatchSnapshot();
});
