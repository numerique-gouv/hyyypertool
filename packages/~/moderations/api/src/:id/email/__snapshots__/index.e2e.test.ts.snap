// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /moderations/:id/email 1`] = `"<!DOCTYPE html><div class="m-auto my-12 flex justify-around"><a href="https://support.etalab.gouv.fr/#search/adora.pony@unicorn.xyz" rel="noopener noreferrer" target="_blank">Trouver l&#39;email correspondant dans Zammad</a><a href="https://app.crisp.chat/website/undefined/inbox/" rel="noopener noreferrer" target="_blank">Trouver l&#39;email correspondant dans Crisp</a></div>"`;
